--- a/Makefile	2018-01-30 19:51:35.000000000 -0600
+++ b/Makefile	2018-03-08 00:49:25.030372595 -0600
@@ -24,7 +24,7 @@
 # BUILD CONFIGURATION OPTIONS:
 #
 # Uncomment next line to build GLES version (embedded systems)
-#USE_GLES=1
+USE_GLES=1
 #
 # Uncomment next line to disable movie support (i.e. no FFmpeg).
 #NO_MOVIE=1
@@ -91,7 +91,7 @@
  
 ifneq ($(origin CROSS),undefined)
 override STRIP := $(TOOLCHAIN)-$(STRIP)
-override PKG_CONFIG := $(TOOLCHAIN)-$(PKG_CONFIG)
+override PKG_CONFIG := $(PKG_CONFIG)
 override WINDRES := $(TOOLCHAIN)-$(WINDRES)
 endif
 
diff --git extlibs/gameswf/gameswf/gameswf_freetype.h extlibs/gameswf/gameswf/gameswf_freetype.h
index 968c0b3..2c5681b 100644
--- a/extlibs/gameswf/gameswf/gameswf_freetype.h
+++ b/extlibs/gameswf/gameswf/gameswf_freetype.h
@@ -16,7 +16,7 @@
 
 #if TU_CONFIG_LINK_TO_FREETYPE == 1
 
-#include <ft2build.h>
+#include <freetype2/ft2build.h>
 #include FT_OUTLINE_H
 #include FT_FREETYPE_H
 #include FT_GLYPH_H
 --- a/src/fe_window.cpp	2018-03-08 01:39:29.183929176 -0600
+++ b/src/fe_window.cpp	2018-03-08 01:39:39.155762102 -0600
@@ -179,10 +179,10 @@
 {
 	int style_map[4] =
 	{
-		sf::Style::None,			// FeSettings::Default
+		sf::Style::None2,			// FeSettings::Default
 		sf::Style::Fullscreen,	// FeSettings::Fullscreen
 		sf::Style::Default,		// FeSettings::Window
-		sf::Style::None			// FeSettings::WindowNoBorder
+		sf::Style::None2			// FeSettings::WindowNoBorder
 	};
 
 	int win_mode = m_fes.get_window_mode();
