From 6b2386e0984885b8719441ffc3b6c8dff1b27d1f Mon Sep 17 00:00:00 2001
From: Adam Green <greena88@gmail.com>
Date: Sun, 18 Mar 2018 00:43:40 +0000
Subject: [PATCH] drivers/amlogic/deinterlace: set noise reduction to off by
 default

---
 drivers/amlogic/deinterlace/nr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/deinterlace/nr.c b/drivers/amlogic/deinterlace/nr.c
index 975514e23440..5f403190b091 100644
--- a/drivers/amlogic/deinterlace/nr.c
+++ b/drivers/amlogic/deinterlace/nr.c
@@ -26,7 +26,7 @@ static bool dnr_reg_update = 1;/*gxtvbb can't work normal,must set to 0*/
 module_param(dnr_reg_update, bool, 0644);
 MODULE_PARM_DESC(dnr_reg_update, "/n dnr dm enable debug /n");
 
-static unsigned int nr2_en = 0x1;
+static unsigned int nr2_en = 0x0;
 module_param_named(nr2_en, nr2_en, uint, 0644);
 
 int global_bs_calc_sw(int *pGbsVldCnt,
