diff --git a/drivers/amlogic/display/osd/osd_fb.c b/drivers/amlogic/display/osd/osd_fb.c
index a783003..f8a6c55 100644
--- a/drivers/amlogic/display/osd/osd_fb.c
+++ b/drivers/amlogic/display/osd/osd_fb.c
@@ -969,6 +969,7 @@ static int osd_open(struct fb_info *info, int arg)
 	fb_index = fbdev->fb_index;
 	fix = &info->fix;
 	var = &info->var;
+	osd_set_afbc(1);
 	if (fb_rmem.base == 0) {
 		pr_info("use ion buffer for fb memory\n");
 		if (!fb_ion_client)
