--- a/src/joystick/linux/SDL_sysjoystick.c	2016-04-14 13:01:21.130522954 +0200
+++ b/src/joystick/linux/SDL_sysjoystick.c	2016-04-14 13:01:28.450421721 +0200
@@ -492,7 +492,7 @@
                 ++joystick->nbuttons;
             }
         }
-        for (i = 0; i < ABS_MAX; ++i) {
+        for (i = 0; i < ABS_MISC; ++i) {
             /* Skip hats */
             if (i == ABS_HAT0X) {
                 i = ABS_HAT3Y;
@@ -753,6 +753,10 @@
                 }
                 break;
             case EV_ABS:
+                if (code >= ABS_MISC) {
+                    break;
+                }
+
                 switch (code) {
                 case ABS_HAT0X:
                 case ABS_HAT0Y:
