diff -Naurp SDL2-2.0.4/cmake/sdlchecks.cmake SDL2-2.0.4-patch/cmake/sdlchecks.cmake
--- SDL2-2.0.4/cmake/sdlchecks.cmake	2016-01-02 21:56:32.000000000 +0200
+++ SDL2-2.0.4-patch/cmake/sdlchecks.cmake	2016-04-20 18:38:21.000000000 +0300
@@ -666,6 +666,29 @@ macro(CheckVivante)
 endmacro(CheckVivante)
 
 # Requires:
+# - n/a
+macro(CheckMali)
+  if(VIDEO_MALI)
+    check_c_source_compiles("
+        #define LINUX
+        #define EGL_API_FB
+        #include <EGL/egl.h>
+        int main(int argc, char** argv) {}" HAVE_VIDEO_MALI_EGL_FB)
+    if(HAVE_VIDEO_MALI_EGL_FB)
+      set(HAVE_VIDEO_MALI TRUE)
+      set(HAVE_SDL_VIDEO TRUE)
+
+      file(GLOB MALI_SOURCES ${SDL2_SOURCE_DIR}/src/video/mali-fbdev/*.c)
+      set(SOURCE_FILES ${SOURCE_FILES} ${MALI_SOURCES})
+      set(SDL_VIDEO_DRIVER_MALI 1)
+      set(SDL_CFLAGS "${SDL_CFLAGS} -DLINUX -DEGL_API_FB")
+      list(APPEND EXTRA_LIBS EGL)
+    endif(HAVE_VIDEO_MALI_EGL_MALI)
+  endif(VIDEO_MALI)
+endmacro(CheckMali)
+
+
+# Requires:
 # - nada
 macro(CheckOpenGLX11)
   if(VIDEO_OPENGL)
diff -Naurp SDL2-2.0.4/CMakeLists.txt SDL2-2.0.4-patch/CMakeLists.txt
--- SDL2-2.0.4/CMakeLists.txt	2016-01-02 21:56:31.000000000 +0200
+++ SDL2-2.0.4-patch/CMakeLists.txt	2016-04-20 18:36:19.000000000 +0300
@@ -293,6 +293,7 @@ set_option(VIDEO_COCOA         "Use Coco
 set_option(DIRECTX             "Use DirectX for Windows audio/video" ${WINDOWS})
 set_option(RENDER_D3D          "Enable the Direct3D render driver" ${WINDOWS})
 set_option(VIDEO_VIVANTE       "Use Vivante EGL video driver" ${UNIX_SYS})
+set_option(VIDEO_MALI          "Use Mali EGL video driver" ${UNIX_SYS})
 
 # TODO: We should (should we?) respect cmake's ${BUILD_SHARED_LIBS} flag here
 # The options below are for compatibility to configure's default behaviour.
@@ -822,6 +823,7 @@ elseif(UNIX AND NOT APPLE)
     CheckOpenGLESX11()
     CheckWayland()
     CheckVivante()
+    CheckMali()
   endif()
 
   if(LINUX)
diff -Naurp SDL2-2.0.4/configure SDL2-2.0.4-patch/configure
--- SDL2-2.0.4/configure	2016-04-20 19:32:44.000000000 +0300
+++ SDL2-2.0.4-patch/configure	2016-04-20 18:42:52.000000000 +0300
@@ -834,6 +834,7 @@ enable_video_x11_scrnsaver
 enable_video_x11_xshape
 enable_video_x11_vm
 enable_video_vivante
+enable_video_mali
 enable_video_cocoa
 enable_video_directfb
 enable_directfb_shared
@@ -1567,6 +1568,7 @@ Optional Features:
                           enable X11 XShape support [[default=yes]]
   --enable-video-x11-vm   use X11 VM extension for fullscreen [[default=yes]]
   --enable-video-vivante  use Vivante EGL video driver [[default=yes]]
+  --enable-video-mali     use Mali EGL video driver [[default=yes]]
   --enable-video-cocoa    use Cocoa video driver [[default=yes]]
   --enable-video-directfb use DirectFB video driver [[default=no]]
   --enable-directfb-shared
@@ -20651,6 +20653,56 @@ $as_echo "#define SDL_VIDEO_DRIVER_VIVAN
     fi
 }
 
+CheckMaliVideo()
+{
+    # Check whether --enable-video-mali was given.
+if test "${enable_video_mali+set}" = set; then :
+  enableval=$enable_video_mali;
+else
+  enable_video_mali=yes
+fi
+
+    if test x$enable_video = xyes -a x$enable_video_mali = xyes; then
+        { $as_echo "$as_me:${as_lineno-$LINENO}: checking for Mali FB API" >&5
+$as_echo_n "checking for Mali FB API... " >&6; }
+        have_mali_egl=no
+        cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+
+          #define LINUX
+          #define EGL_API_FB
+          #include <EGL/egl.h>
+
+int
+main ()
+{
+
+
+  ;
+  return 0;
+}
+_ACEOF
+if ac_fn_c_try_compile "$LINENO"; then :
+
+        have_mali_egl=yes
+
+fi
+rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
+        { $as_echo "$as_me:${as_lineno-$LINENO}: result: $have_mali_egl" >&5
+$as_echo "$have_mali_egl" >&6; }
+
+        if test x$have_mali_egl = xyes; then
+
+$as_echo "#define SDL_VIDEO_DRIVER_MALI 1" >>confdefs.h
+
+            EXTRA_CFLAGS="$EXTRA_CFLAGS -DLINUX -DEGL_API_FB"
+            SOURCES="$SOURCES $srcdir/src/video/mali-fbdev/*.c"
+            SUMMARY_video="${SUMMARY_video} mali"
+            have_video=yes
+        fi
+    fi
+}
+
 CheckHaikuVideo()
 {
     if test x$enable_video = xyes; then
@@ -22996,6 +23048,7 @@ case "$host" in
         CheckLinuxVersion
         CheckRPATH
         CheckVivanteVideo
+        CheckMaliVideo
         # Set up files for the audio library
         if test x$enable_audio = xyes; then
           case $ARCH in
diff -Naurp SDL2-2.0.4/configure.in SDL2-2.0.4-patch/configure.in
--- SDL2-2.0.4/configure.in	2016-01-02 21:56:32.000000000 +0200
+++ SDL2-2.0.4-patch/configure.in	2016-04-20 18:46:19.000000000 +0300
@@ -1754,6 +1754,35 @@ AC_HELP_STRING([--enable-video-vivante],
     fi
 }
 
+dnl Set up the Mali video driver if enabled
+CheckMaliVideo()
+{
+    AC_ARG_ENABLE(video-mali,
+AC_HELP_STRING([--enable-video-mali], [use Mali EGL video driver [[default=yes]]]),
+                  , enable_video_mali=yes)
+    if test x$enable_video = xyes -a x$enable_video_mali = xyes; then
+        AC_MSG_CHECKING(for Mali FB API)
+        have_mali_egl=no
+        AC_TRY_COMPILE([
+          #define LINUX
+          #define EGL_API_FB
+          #include <EGL/egl.h>
+        ],[
+        ],[
+        have_mali_egl=yes
+        ])
+        AC_MSG_RESULT($have_mali_egl)
+
+        if test x$have_mali_egl = xyes; then
+            AC_DEFINE(SDL_VIDEO_DRIVER_MALI, 1, [ ])
+            EXTRA_CFLAGS="$EXTRA_CFLAGS -DLINUX -DEGL_API_FB"
+            SOURCES="$SOURCES $srcdir/src/video/mali-fbdev/*.c"
+            SUMMARY_video="${SUMMARY_video} mali"
+            have_video=yes
+        fi
+    fi
+}
+
 dnl Set up the Haiku video driver if enabled
 CheckHaikuVideo()
 {
@@ -2904,6 +2933,7 @@ case "$host" in
         CheckLinuxVersion
         CheckRPATH
         CheckVivanteVideo
+        CheckMaliVideo
         # Set up files for the audio library
         if test x$enable_audio = xyes; then
           case $ARCH in
diff -Naurp SDL2-2.0.4/include/SDL_config.h.cmake SDL2-2.0.4-patch/include/SDL_config.h.cmake
--- SDL2-2.0.4/include/SDL_config.h.cmake	2016-01-02 21:56:31.000000000 +0200
+++ SDL2-2.0.4-patch/include/SDL_config.h.cmake	2016-04-20 18:51:09.000000000 +0300
@@ -281,6 +281,7 @@
 #cmakedefine SDL_VIDEO_DRIVER_RPI @SDL_VIDEO_DRIVER_RPI@
 #cmakedefine SDL_VIDEO_DRIVER_VIVANTE @SDL_VIDEO_DRIVER_VIVANTE@
 #cmakedefine SDL_VIDEO_DRIVER_VIVANTE_VDK @SDL_VIDEO_DRIVER_VIVANTE_VDK@
+#cmakedefine SDL_VIDEO_DRIVER_MALI @SDL_VIDEO_DRIVER_MALI@
 
 #cmakedefine SDL_VIDEO_DRIVER_WAYLAND_QT_TOUCH @SDL_VIDEO_DRIVER_WAYLAND_QT_TOUCH@
 #cmakedefine SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC @SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC@
diff -Naurp SDL2-2.0.4/include/SDL_config.h.in SDL2-2.0.4-patch/include/SDL_config.h.in
--- SDL2-2.0.4/include/SDL_config.h.in	2016-01-02 21:56:31.000000000 +0200
+++ SDL2-2.0.4-patch/include/SDL_config.h.in	2016-04-20 18:52:16.000000000 +0300
@@ -288,6 +288,7 @@
 #undef SDL_VIDEO_DRIVER_MIR_DYNAMIC_XKBCOMMON
 #undef SDL_VIDEO_DRIVER_X11
 #undef SDL_VIDEO_DRIVER_RPI
+#undef SDL_VIDEO_DRIVER_MALI
 #undef SDL_VIDEO_DRIVER_ANDROID
 #undef SDL_VIDEO_DRIVER_EMSCRIPTEN
 #undef SDL_VIDEO_DRIVER_X11_DYNAMIC
diff -Naurp SDL2-2.0.4/src/video/SDL_egl.c SDL2-2.0.4-patch/src/video/SDL_egl.c
--- SDL2-2.0.4/src/video/SDL_egl.c	2016-04-20 19:32:44.000000000 +0300
+++ SDL2-2.0.4-patch/src/video/SDL_egl.c	2016-04-20 18:54:35.000000000 +0300
@@ -45,8 +45,8 @@
 #define DEFAULT_OGL_ES_PVR "/usr/lib/libGLESv2.so"
 #define DEFAULT_OGL_ES "/usr/lib/libGLESv2.so"
 
-#elif SDL_VIDEO_DRIVER_ANDROID || SDL_VIDEO_DRIVER_VIVANTE
-/* Android */
+#elif SDL_VIDEO_DRIVER_ANDROID || SDL_VIDEO_DRIVER_VIVANTE || SDL_VIDEO_DRIVER_MALI
+/* Android and Mali-fbdev */
 #define DEFAULT_EGL "libEGL.so"
 #define DEFAULT_OGL_ES2 "libGLESv2.so"
 #define DEFAULT_OGL_ES_PVR "libGLES_CM.so"
diff -Naurp SDL2-2.0.4/src/video/SDL_sysvideo.h SDL2-2.0.4-patch/src/video/SDL_sysvideo.h
--- SDL2-2.0.4/src/video/SDL_sysvideo.h	2016-01-02 21:56:31.000000000 +0200
+++ SDL2-2.0.4-patch/src/video/SDL_sysvideo.h	2016-04-20 18:55:52.000000000 +0300
@@ -389,6 +389,9 @@ extern VideoBootStrap PSP_bootstrap;
 #if SDL_VIDEO_DRIVER_RPI
 extern VideoBootStrap RPI_bootstrap;
 #endif
+#if SDL_VIDEO_DRIVER_MALI
+extern VideoBootStrap MALI_bootstrap;
+#endif
 #if SDL_VIDEO_DRIVER_DUMMY
 extern VideoBootStrap DUMMY_bootstrap;
 #endif
diff -Naurp SDL2-2.0.4/src/video/SDL_video.c SDL2-2.0.4-patch/src/video/SDL_video.c
--- SDL2-2.0.4/src/video/SDL_video.c	2016-01-02 21:56:31.000000000 +0200
+++ SDL2-2.0.4-patch/src/video/SDL_video.c	2016-04-20 18:57:03.000000000 +0300
@@ -98,6 +98,9 @@ static VideoBootStrap *bootstrap[] = {
 #endif
 #if SDL_VIDEO_DRIVER_RPI
     &RPI_bootstrap,
+#endif
+#if SDL_VIDEO_DRIVER_MALI
+    &MALI_bootstrap,
 #endif 
 #if SDL_VIDEO_DRIVER_NACL
     &NACL_bootstrap,
