--- a/src/backend/Backend.cpp	2018-04-20 20:20:14.894906639 -0500
+++ b/src/backend/Backend.cpp	2018-04-21 00:43:11.880222519 -0500
@@ -31,8 +31,10 @@
 {
     using ScriptEvent = ScriptRunner::EventType;
 
-    ScriptRunner::findAndRunScripts(ScriptEvent::QUIT);
     switch (type) {
+       case AppCloseType::QUIT:
+            ScriptRunner::findAndRunScripts(ScriptEvent::QUIT);
+            break;
         case AppCloseType::REBOOT:
             ScriptRunner::findAndRunScripts(ScriptEvent::REBOOT);
             break;
