--- ./configure.ac	2017-03-02 20:59:08.648690810 -0600
+++ ./configure.ac	2017-03-03 14:36:42.135407995 -0600
@@ -779,7 +779,7 @@
 if test $ac_lib_freetype = auto; then
 	AC_PATH_PROGS(
 		[ac_lib_freetype_config],
-		[${ac_tool_prefix}freetype-config],
+		[freetype-config],
 		[],
 		[${ac_lib_freetype_path}]
 	)
@@ -814,12 +814,12 @@
 elif test $ac_lib_freetype = yes; then
 	AC_PATH_PROGS(
 		[ac_lib_freetype_config],
-		[${ac_tool_prefix}freetype-config],
+		[freetype-config],
 		[],
 		[${ac_lib_freetype_path}]
 	)
 	if test -z $ac_lib_freetype_config ; then
-		AC_MSG_ERROR([the ${ac_tool_prefix}freetype-config tool is missing])
+		AC_MSG_ERROR([the freetype-config tool is missing])
 	else
 		ac_save_CFLAGS="$CFLAGS"
 		ac_save_LIBS="$LIBS"
@@ -994,7 +994,7 @@
 		else
 			AC_PATH_PROGS(
 				[ac_lib_sdl_config],
-				[${ac_tool_prefix}sdl-config],
+				[sdl-config],
 				[],
 				[${ac_lib_sdl_path}]
 			)
@@ -1030,12 +1030,12 @@
 	fi
 	AC_PATH_PROGS(
 		[ac_lib_sdl_config],
-		[${ac_tool_prefix}sdl-config],
+		[sdl-config],
 		[],
 		[${ac_lib_sdl_path}]
 	)
 	if test -z $ac_lib_sdl_config ; then
-		AC_MSG_ERROR([the ${ac_tool_prefix}sdl-config tool is missing])
+		AC_MSG_ERROR([the sdl-config tool is missing])
 	else
 		ac_save_CFLAGS="$CFLAGS"
 		ac_save_LIBS="$LIBS"
