diff -Naur kodi-16.1-c327c53.orig/xbmc/utils/CPUInfo.cpp kodi-16.1-c327c53/xbmc/utils/CPUInfo.cpp
--- kodi-16.1-c327c53.orig/xbmc/utils/CPUInfo.cpp	2016-06-17 14:03:24.527466671 +0200
+++ kodi-16.1-c327c53/xbmc/utils/CPUInfo.cpp	2016-06-17 14:04:57.509159757 +0200
@@ -624,7 +624,6 @@
     if (!ret)
     {
       ret = fscanf(m_fProcTemperature, "%d", &value);
-      value = value / 1000;
       scale = 'c';
       ret++;
     }
