diff -Naurp kodi-17.0-beta7-b9fe8e9/xbmc/powermanagement/linux/LogindUPowerSyscall.cpp kodi-17.0-beta7-b9fe8e9-patch/xbmc/powermanagement/linux/LogindUPowerSyscall.cpp
--- kodi-17.0-beta7-b9fe8e9/xbmc/powermanagement/linux/LogindUPowerSyscall.cpp	2016-12-01 11:20:47.000000000 +0200
+++ kodi-17.0-beta7-b9fe8e9-patch/xbmc/powermanagement/linux/LogindUPowerSyscall.cpp	2016-12-05 20:47:53.000000000 +0200
@@ -50,10 +50,10 @@ CLogindUPowerSyscall::CLogindUPowerSysca
   if (!m_hasUPower)
     CLog::Log(LOGINFO, "LogindUPowerSyscall - UPower not found, battery information will not be available");
 
-  m_canPowerdown = LogindCheckCapability("CanPowerOff");
-  m_canReboot    = LogindCheckCapability("CanReboot");
-  m_canHibernate = false;
-  m_canSuspend   = LogindCheckCapability("CanSuspend");
+  m_canPowerdown = true;
+  m_canReboot    = true;
+  m_canHibernate = true;
+  m_canSuspend   = true;
 
   InhibitDelayLock();
 

