diff --git a/drivers/amlogic/cec/hdmi_ao_cec.c b/drivers/amlogic/cec/hdmi_ao_cec.c
index a40650d04cb2..752205851529 100644
--- a/drivers/amlogic/cec/hdmi_ao_cec.c
+++ b/drivers/amlogic/cec/hdmi_ao_cec.c
@@ -1531,12 +1531,6 @@ static void cec_task(struct work_struct *work)
 	dwork = &cec_dev->cec_work;
 	if (cec_dev &&
 			!(cec_dev->hal_flag & (1 << HDMI_OPTION_SYSTEM_CEC_CONTROL))) {
-		if (1 << cec_dev->cec_info.log_addr & (1 << 0x0 | 1 << 0xF)) {
-			ret = cec_node_init(cec_dev->tx_dev);
-			if (ret < 0) {
-				return;
-			}
-		}
 		cec_rx_process();
 	}
 	if (!ee_cec && !cec_late_check_rx_buffer())
