diff --git a/drivers/amlogic/hdmi/hdmi_tx_20/hw/hdmi_tx_hw.c b/drivers/amlogic/hdmi/hdmi_tx_20/hw/hdmi_tx_hw.c
index ba549c7..18e0c48 100644
--- a/drivers/amlogic/hdmi/hdmi_tx_20/hw/hdmi_tx_hw.c
+++ b/drivers/amlogic/hdmi/hdmi_tx_20/hw/hdmi_tx_hw.c
@@ -510,6 +510,7 @@ static irqreturn_t intr_handler(int irq, void *dev)
 			&hdev->work_hpd_plugin, HZ / 3);
 	}
 	/* HPD falling */
+	/*
 	if (data32 & (1 << 2)) {
 		hdev->hdmitx_event |= HDMI_TX_HPD_PLUGOUT;
 		hdev->hdmitx_event &= ~HDMI_TX_HPD_PLUGIN;
@@ -518,6 +519,7 @@ static irqreturn_t intr_handler(int irq, void *dev)
 		queue_delayed_work(hdev->hdmi_wq,
 			&hdev->work_hpd_plugout, 0);
 	}
+	*/
 next:
 	/* internal interrupt */
 	if (data32 & (1 << 0)) {
@@ -2289,6 +2291,13 @@ static void hdmitx_setupirq(struct hdmitx_dev *phdev)
 	r = request_irq(phdev->irq_hpd, &intr_handler,
 			IRQF_SHARED, "hdmitx",
 			(void *)phdev);
+
+	phdev->hdmitx_event |= HDMI_TX_HPD_PLUGIN;
+	phdev->hdmitx_event &= ~HDMI_TX_HPD_PLUGOUT;
+	PREPARE_DELAYED_WORK(&phdev->work_hpd_plugin,
+		hdmitx_hpd_plugin_handler);
+	queue_delayed_work(phdev->hdmi_wq,
+		&phdev->work_hpd_plugin, HZ/3);
 }
 
 static void hdmitx_uninit(struct hdmitx_dev *phdev)
