diff --git a/arch/arm64/boot/dts/amlogic/gxl_p212_1g.dts b/arch/arm64/boot/dts/amlogic/gxl_p212_1g.dts
index aaba8a4..45ef336 100644
--- a/arch/arm64/boot/dts/amlogic/gxl_p212_1g.dts
+++ b/arch/arm64/boot/dts/amlogic/gxl_p212_1g.dts
@@ -951,13 +951,13 @@
 	dummy_codec:dummy{
 		#sound-dai-cells = <0>;
 		compatible = "amlogic, aml_dummy_codec";
-		status = "disable";
+		status = "okay";
 	};
 	amlogic_codec:t9015{
 		#sound-dai-cells = <0>;
 		compatible = "amlogic, aml_codec_T9015";
 		reg = <0x0 0xc8832000 0x0 0x14>;
-		status = "okay";
+		status = "disable";
 	};
 	aml_m8_snd {
 		compatible = "aml, aml_snd_m8";
@@ -988,7 +988,7 @@
 			sound-dai = <&pcm_dai>;
 		};
 		codec0: codec0 {
-			sound-dai = <&amlogic_codec>;
+			sound-dai = <&dummy_codec>;
 		};
 		codec1: codec1 {
 			sound-dai = <&spdif_codec>;
diff --git a/arch/arm64/boot/dts/amlogic/gxl_p212_2g.dts b/arch/arm64/boot/dts/amlogic/gxl_p212_2g.dts
index d1c93ed..96a0465 100644
--- a/arch/arm64/boot/dts/amlogic/gxl_p212_2g.dts
+++ b/arch/arm64/boot/dts/amlogic/gxl_p212_2g.dts
@@ -952,13 +952,13 @@
 	dummy_codec:dummy{
 		#sound-dai-cells = <0>;
 		compatible = "amlogic, aml_dummy_codec";
-		status = "disable";
+		status = "okay";
 	};
 	amlogic_codec:t9015{
 		#sound-dai-cells = <0>;
 		compatible = "amlogic, aml_codec_T9015";
 		reg = <0x0 0xc8832000 0x0 0x14>;
-		status = "okay";
+		status = "disable";
 	};
 	aml_m8_snd {
 		compatible = "aml, aml_snd_m8";
@@ -989,7 +989,7 @@
 			sound-dai = <&pcm_dai>;
 		};
 		codec0: codec0 {
-			sound-dai = <&amlogic_codec>;
+			sound-dai = <&dummy_codec>;
 		};
 		codec1: codec1 {
 			sound-dai = <&spdif_codec>;
diff --git a/arch/arm64/boot/dts/amlogic/gxl_p230_2g.dts b/arch/arm64/boot/dts/amlogic/gxl_p230_2g.dts
index 2b988d5..f8b7f44 100644
--- a/arch/arm64/boot/dts/amlogic/gxl_p230_2g.dts
+++ b/arch/arm64/boot/dts/amlogic/gxl_p230_2g.dts
@@ -974,13 +974,13 @@
 	dummy_codec:dummy{
 		#sound-dai-cells = <0>;
 		compatible = "amlogic, aml_dummy_codec";
-		status = "disable";
+		status = "okay";
 	};
 	amlogic_codec:t9015{
 		#sound-dai-cells = <0>;
 		compatible = "amlogic, aml_codec_T9015";
 		reg = <0x0 0xc8832000 0x0 0x14>;
-		status = "okay";
+		status = "disable";
 	};
 	aml_m8_snd {
 		compatible = "aml, aml_snd_m8";
@@ -1011,7 +1011,7 @@
 			sound-dai = <&pcm_dai>;
 		};
 		codec0: codec0 {
-			sound-dai = <&amlogic_codec>;
+			sound-dai = <&dummy_codec>;
 		};
 		codec1: codec1 {
 			sound-dai = <&spdif_codec>;
diff --git a/arch/arm64/boot/dts/amlogic/gxl_p231_1g.dts b/arch/arm64/boot/dts/amlogic/gxl_p231_1g.dts
index 7b1f03c..a8959de 100644
--- a/arch/arm64/boot/dts/amlogic/gxl_p231_1g.dts
+++ b/arch/arm64/boot/dts/amlogic/gxl_p231_1g.dts
@@ -951,13 +951,13 @@
 	dummy_codec:dummy{
 		#sound-dai-cells = <0>;
 		compatible = "amlogic, aml_dummy_codec";
-		status = "disable";
+		status = "okay";
 	};
 	amlogic_codec:t9015{
 		#sound-dai-cells = <0>;
 		compatible = "amlogic, aml_codec_T9015";
 		reg = <0x0 0xc8832000 0x0 0x14>;
-		status = "okay";
+		status = "disable";
 	};
 	aml_m8_snd {
 		compatible = "aml, aml_snd_m8";
@@ -988,7 +988,7 @@
 			sound-dai = <&pcm_dai>;
 		};
 		codec0: codec0 {
-			sound-dai = <&amlogic_codec>;
+			sound-dai = <&dummy_codec>;
 		};
 		codec1: codec1 {
 			sound-dai = <&spdif_codec>;
diff --git a/arch/arm64/boot/dts/amlogic/gxl_p231_2g.dts b/arch/arm64/boot/dts/amlogic/gxl_p231_2g.dts
index 93c11a6..1621639 100644
--- a/arch/arm64/boot/dts/amlogic/gxl_p231_2g.dts
+++ b/arch/arm64/boot/dts/amlogic/gxl_p231_2g.dts
@@ -951,13 +951,13 @@
 	dummy_codec:dummy{
 		#sound-dai-cells = <0>;
 		compatible = "amlogic, aml_dummy_codec";
-		status = "disable";
+		status = "okay";
 	};
 	amlogic_codec:t9015{
 		#sound-dai-cells = <0>;
 		compatible = "amlogic, aml_codec_T9015";
 		reg = <0x0 0xc8832000 0x0 0x14>;
-		status = "okay";
+		status = "disable";
 	};
 	aml_m8_snd {
 		compatible = "aml, aml_snd_m8";
@@ -988,7 +988,7 @@
 			sound-dai = <&pcm_dai>;
 		};
 		codec0: codec0 {
-			sound-dai = <&amlogic_codec>;
+			sound-dai = <&dummy_codec>;
 		};
 		codec1: codec1 {
 			sound-dai = <&spdif_codec>;
diff --git a/arch/arm64/boot/dts/amlogic/gxl_pxp.dts b/arch/arm64/boot/dts/amlogic/gxl_pxp.dts
index 62e02d3..e4c709c 100644
--- a/arch/arm64/boot/dts/amlogic/gxl_pxp.dts
+++ b/arch/arm64/boot/dts/amlogic/gxl_pxp.dts
@@ -903,13 +903,13 @@
 	dummy_codec:dummy{
 		#sound-dai-cells = <0>;
 		compatible = "amlogic, aml_dummy_codec";
-		status = "disable";
+		status = "okay";
 	};
 	amlogic_codec:t9015{
 		#sound-dai-cells = <0>;
 		compatible = "amlogic, aml_codec_T9015";
 		reg = <0x0 0xc8832000 0x0 0x14>;
-		status = "okay";
+		status = "disable";
 	};
 	aml_m8_snd {
 		compatible = "aml, aml_snd_m8";
@@ -938,7 +938,7 @@
 			sound-dai = <&pcm_dai>;
 		};
 		codec0: codec0 {
-			sound-dai = <&amlogic_codec>;
+			sound-dai = <&dummy_codec>;
 		};
 		codec1: codec1 {
 			sound-dai = <&spdif_codec>;
diff --git a/arch/arm64/boot/dts/amlogic/gxl_skt.dts b/arch/arm64/boot/dts/amlogic/gxl_skt.dts
index 1a2737f..5afce33 100644
--- a/arch/arm64/boot/dts/amlogic/gxl_skt.dts
+++ b/arch/arm64/boot/dts/amlogic/gxl_skt.dts
@@ -930,13 +930,13 @@
 	dummy_codec:dummy{
 		#sound-dai-cells = <0>;
 		compatible = "amlogic, aml_dummy_codec";
-		status = "disable";
+		status = "okay";
 	};
 	amlogic_codec:t9015{
 		#sound-dai-cells = <0>;
 		compatible = "amlogic, aml_codec_T9015";
 		reg = <0x0 0xc8832000 0x0 0x14>;
-		status = "okay";
+		status = "disable";
 	};
 	aml_m8_snd {
 		compatible = "aml, aml_snd_m8";
@@ -966,7 +966,7 @@
 			sound-dai = <&pcm_dai>;
 		};
 		codec0: codec0 {
-			sound-dai = <&amlogic_codec>;
+			sound-dai = <&dummy_codec>;
 		};
 		codec1: codec1 {
 			sound-dai = <&spdif_codec>;
