--- libretro-ppsspp-5f7bcf7/ffmpeg/linux_arm.sh	2017-02-25 11:29:12.166940037 -0600
+++ libretro-ppsspp-5f7bcf7/ffmpeg/linux_arm.sh	2017-02-25 11:29:32.346949412 -0600
@@ -78,7 +78,7 @@
 ./configure --target-os=linux \
     --prefix=./linux/armv6 \
     ${GENERAL} \
-    --extra-cflags=" -O3 -fasm -Wno-psabi -fno-short-enums -fno-strict-aliasing -finline-limit=300 -mfloat-abi=softfp -mfpu=vfp -marm -march=armv6" \
+    --extra-cflags=" -O3 -fPIC -fasm -Wno-psabi -fno-short-enums -fno-strict-aliasing -finline-limit=300 -mfloat-abi=softfp -mfpu=vfp -marm -march=armv6" \
     --disable-shared \
     --enable-static \
     --enable-zlib \
@@ -103,7 +103,7 @@
 ./configure --target-os=linux \
     --prefix=./linux/armv7 \
     ${GENERAL} \
-    --extra-cflags=" -O3 -fasm -Wno-psabi -fno-short-enums -fno-strict-aliasing -finline-limit=300 -mfloat-abi=softfp -mfpu=neon -marm -march=armv7-a" \
+    --extra-cflags=" -O3 -fPIC -fasm -Wno-psabi -fno-short-enums -fno-strict-aliasing -finline-limit=300 -mfloat-abi=softfp -mfpu=neon -marm -march=armv7-a" \
     --disable-shared \
     --enable-static \
     --enable-zlib \
