From 9d56d5ed0173e1a61b123dee46fa4c9673b2111b Mon Sep 17 00:00:00 2001
From: kszaq <kszaquitto@gmail.com>
Date: Tue, 19 Sep 2017 22:11:49 +0200
Subject: [PATCH 4/4] bcmdhd_1_201_59_x: add BCM4359C0 firmware loading support

---
 bcmdhd_1_201_59_x/dhd_config.c | 12 ++++++++++++
 bcmdhd_1_201_59_x/dhd_config.h |  1 +
 2 files changed, 13 insertions(+)

diff --git a/bcmdhd_1_201_59_x/dhd_config.c b/bcmdhd_1_201_59_x/dhd_config.c
index f8363cf..fafce29 100644
--- a/bcmdhd_1_201_59_x/dhd_config.c
+++ b/bcmdhd_1_201_59_x/dhd_config.c
@@ -60,6 +60,7 @@ uint config_msg_level = CONFIG_ERROR_LEVEL;
 #define BCM4354A1_CONF_NAME "config_4354a1.txt"
 #define BCM4356A2_CONF_NAME "config_4356a2.txt"
 #define BCM4359B1_CONF_NAME "config_4359b1.txt"
+#define BCM4359C0_CONF_NAME "config_4359c0.txt"
 #endif
 
 #ifdef BCMSDIO
@@ -170,6 +171,13 @@ const static char *bcm4359b1_ag_fw_name[] = {
 	"fw_bcm4359b1_ag_p2p.bin",
 	"fw_bcm4359b1_ag_mfg.bin"
 };
+
+const static char *bcm4359c0_ag_fw_name[] = {
+	"fw_bcm4359c0_ag.bin",
+	"fw_bcm4359c0_ag_apsta.bin",
+	"fw_bcm4359c0_ag_p2p.bin",
+	"fw_bcm4359c0_ag_mfg.bin"
+};
 #endif
 #ifdef BCMPCIE
 const static char *bcm4356a2_pcie_ag_fw_name[] = {
@@ -535,6 +543,8 @@ dhd_conf_set_fw_name_by_chip(dhd_pub_t *dhd, char *fw_path)
 		case BCM4359_CHIP_ID:
 			if (chiprev == BCM4359B1_CHIP_REV)
 				strcpy(&fw_path[i+1], bcm4359b1_ag_fw_name[fw_type]);
+			else if (chiprev == BCM4359C0_CHIP_REV)
+				strcpy(&fw_path[i+1], bcm4359c0_ag_fw_name[fw_type]);
 			break;
 #endif
 #ifdef BCMPCIE
@@ -699,6 +709,8 @@ dhd_conf_set_conf_name_by_chip(dhd_pub_t *dhd, char *conf_path)
 		case BCM4359_CHIP_ID:
 			if (chiprev == BCM4359B1_CHIP_REV)
 				strcpy(&conf_path[i+1], BCM4359B1_CONF_NAME);
+			else if (chiprev == BCM4359C0_CHIP_REV)
+				strcpy(&conf_path[i+1], BCM4359C0_CONF_NAME);
 			break;
 #endif
 #ifdef BCMPCIE
diff --git a/bcmdhd_1_201_59_x/dhd_config.h b/bcmdhd_1_201_59_x/dhd_config.h
index 3be976a..3bd9b52 100644
--- a/bcmdhd_1_201_59_x/dhd_config.h
+++ b/bcmdhd_1_201_59_x/dhd_config.h
@@ -33,6 +33,7 @@ extern uint dhd_slpauto;
 #define BCM43455C0_CHIP_REV     6
 #define BCM4354A1_CHIP_REV      1
 #define BCM4359B1_CHIP_REV      5
+#define BCM4359C0_CHIP_REV      9
 #endif
 #define BCM4356A2_CHIP_REV      2
 
-- 
2.7.4

