From 7ec9dfb21e17798f0434414a6bb1b8406b174352 Mon Sep 17 00:00:00 2001
From: kszaq <kszaquitto@gmail.com>
Date: Thu, 14 Sep 2017 21:20:18 +0200
Subject: [PATCH 1/4] bcmdhd: Add new SDIO vendor/device ID for AP6212

---
 bcmdhd_1_201_59_x/bcmsdh_sdmmc_linux.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/bcmdhd_1_201_59_x/bcmsdh_sdmmc_linux.c b/bcmdhd_1_201_59_x/bcmsdh_sdmmc_linux.c
index 19e0018..007eae8 100644
--- a/bcmdhd_1_201_59_x/bcmsdh_sdmmc_linux.c
+++ b/bcmdhd_1_201_59_x/bcmsdh_sdmmc_linux.c
@@ -71,6 +71,9 @@
 #if !defined(SDIO_DEVICE_ID_BROADCOM_43239)
 #define SDIO_DEVICE_ID_BROADCOM_43239    43239
 #endif /* !defined(SDIO_DEVICE_ID_BROADCOM_43239) */
+#if !defined(SDIO_DEVICE_ID_BROADCOM_4343)
+#define SDIO_DEVICE_ID_BROADCOM_4343   0xa9a6
+#endif /* !defined(SDIO_DEVICE_ID_BROADCOM_4343) */
 
 extern void wl_cfg80211_set_parent_dev(void *dev);
 extern void sdioh_sdmmc_devintr_off(sdioh_info_t *sd);
@@ -211,6 +214,7 @@ static void bcmsdh_sdmmc_remove(struct sdio_func *func)
 /* devices we support, null terminated */
 static const struct sdio_device_id bcmsdh_sdmmc_ids[] = {
 	{ SDIO_DEVICE(SDIO_VENDOR_ID_BROADCOM, SDIO_DEVICE_ID_BROADCOM_DEFAULT) },
+	{ SDIO_DEVICE(SDIO_VENDOR_ID_BROADCOM, SDIO_DEVICE_ID_BROADCOM_4343) },
 	{ SDIO_DEVICE(SDIO_VENDOR_ID_BROADCOM, SDIO_DEVICE_ID_BROADCOM_4325_SDGWB) },
 	{ SDIO_DEVICE(SDIO_VENDOR_ID_BROADCOM, SDIO_DEVICE_ID_BROADCOM_4325) },
 	{ SDIO_DEVICE(SDIO_VENDOR_ID_BROADCOM, SDIO_DEVICE_ID_BROADCOM_4329) },
-- 
2.7.4

