From dbd86387683b19963c89f7a17653cc57e40d9d89 Mon Sep 17 00:00:00 2001
From: kszaq <kszaquitto@gmail.com>
Date: Thu, 14 Sep 2017 21:20:55 +0200
Subject: [PATCH 2/4] bcmdhd: add BCM4335 firmware loading support

---
 bcmdhd_1_201_59_x/dhd_config.c | 10 ++++++++++
 bcmdhd_1_201_59_x/dhd_config.h |  1 +
 2 files changed, 11 insertions(+)

diff --git a/bcmdhd_1_201_59_x/dhd_config.c b/bcmdhd_1_201_59_x/dhd_config.c
index 376c041..f8363cf 100644
--- a/bcmdhd_1_201_59_x/dhd_config.c
+++ b/bcmdhd_1_201_59_x/dhd_config.c
@@ -128,6 +128,14 @@ const static char *bcm43241b4_ag_fw_name[] = {
 	"fw_bcm43241b4_ag_mfg.bin"
 };
 
+const static char *bcm4335b0_ag_fw_name[] = {
+	"fw_bcm4335b0_ag.bin",
+	"fw_bcm4335b0_ag_apsta.bin",
+	"fw_bcm4335b0_ag_p2p.bin",
+	"fw_bcm4335b0_ag_es.bin",
+	"fw_bcm4335b0_ag_mfg.bin"
+};
+
 const static char *bcm4339a0_ag_fw_name[] = {
 	"fw_bcm4339a0_ag.bin",
 	"fw_bcm4339a0_ag_apsta.bin",
@@ -501,6 +509,8 @@ dhd_conf_set_fw_name_by_chip(dhd_pub_t *dhd, char *fw_path)
 		case BCM4335_CHIP_ID:
 			if (chiprev == BCM4335A0_CHIP_REV)
 				strcpy(&fw_path[i+1], bcm4339a0_ag_fw_name[fw_type]);
+			else if (chiprev == BCM4335B0_CHIP_REV)
+				strcpy(&fw_path[i+1], bcm4335b0_ag_fw_name[fw_type]);
 			break;
 		case BCM4345_CHIP_ID:
 		case BCM43454_CHIP_ID:
diff --git a/bcmdhd_1_201_59_x/dhd_config.h b/bcmdhd_1_201_59_x/dhd_config.h
index 5437016..3be976a 100644
--- a/bcmdhd_1_201_59_x/dhd_config.h
+++ b/bcmdhd_1_201_59_x/dhd_config.h
@@ -29,6 +29,7 @@ extern uint dhd_slpauto;
 #define BCM43241B4_CHIP_REV     5
 #define BCM4335A0_CHIP_REV      2
 #define BCM4339A0_CHIP_REV      1
+#define BCM4335B0_CHIP_REV      1
 #define BCM43455C0_CHIP_REV     6
 #define BCM4354A1_CHIP_REV      1
 #define BCM4359B1_CHIP_REV      5
-- 
2.7.4

