--- a/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp	2017-04-21 14:42:16.371525222 +0200
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp	2017-04-21 14:43:32.295525222 +0200
@@ -1315,8 +1315,8 @@
       // renderer/drivers have internal queues, being slightliy late here does not mean that
       // we are really late. The likelihood that we recover decreases the greater m_lateframes
       // get. Skipping a frame is easier than having decoder dropping one (lateframes > 10)
-      double x = (m_lateframes <= 6) ? 0.98 : 0;
+      double x = (m_lateframes <= 6) ? 2.0 : 0;
       if (renderPts < m_Queue[*iter].pts + x * frametime)
         break;
       idx = *iter;
       ++iter;
