diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
index 56496e0..9c9ecf2 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
@@ -1989,12 +1989,9 @@ int CAMLCodec::Decode(uint8_t *pData, size_t iSize, double dts, double pts)
     if (loop == 100)
       // Decoder got stuck; Reset
       Reset();
-
-    if ((m_state & STATE_PREFILLED) == 0 && timesize >= 1.0)
-       m_state |= STATE_PREFILLED;
   }
 
-  if ((m_state & STATE_PREFILLED) != 0 && timesize > 0.5 &&  DequeueBuffer() == 0)
+  if (timesize > 0.0 && DequeueBuffer() == 0)
     rtn |= VC_PICTURE;
 
   if (((rtn & VC_PICTURE) == 0 && timesize < 2.0) || timesize < 1.0)
