diff --git a/script.moonlight/lib/moonlight.py b/script.moonlight/lib/moonlight.py
index 7edae39..b526873 100644
--- a/script.moonlight/lib/moonlight.py
+++ b/script.moonlight/lib/moonlight.py
@@ -37,8 +37,8 @@ class _HTTP_DATA(ctypes.Structure):
 
 class LibGameStream:
     def __init__(self, libpath = ""):
-        self.commonlib = ctypes.cdll.LoadLibrary(os.path.join(libpath, "libmoonlight-common.so.0"))
-        self.gslib = ctypes.cdll.LoadLibrary(os.path.join(libpath, "libgamestream.so.0"))
+        self.commonlib = ctypes.cdll.LoadLibrary(os.path.join(libpath, "libmoonlight-common.so"))
+        self.gslib = ctypes.cdll.LoadLibrary(os.path.join(libpath, "libgamestream.so"))
         self.connected = False
         self.address = ""
         self.key_dir = ""
