diff --git a/media/drivers/media/usb/dvb-usb/dibusb-mc-common.c b/media/drivers/media/usb/dvb-usb/dibusb-mc-common.c
index c989cac..8f4aad4 100644
--- a/media/drivers/media/usb/dvb-usb/dibusb-mc-common.c
+++ b/media/drivers/media/usb/dvb-usb/dibusb-mc-common.c
@@ -165,3 +165,6 @@ int dibusb_dib3000mc_tuner_attach(struct dvb_usb_adapter *adap)
 	return 0;
 }
 EXPORT_SYMBOL(dibusb_dib3000mc_tuner_attach);
+
+MODULE_VERSION("1.0");
+MODULE_LICENSE("GPL");
