diff -Naurp emulationstation-c6bfeb8/CMakeLists.txt emulationstation-c6bfeb8-patch/CMakeLists.txt
--- emulationstation-c6bfeb8/CMakeLists.txt	2016-04-20 20:08:47.000000000 +0300
+++ emulationstation-c6bfeb8-patch/CMakeLists.txt	2016-04-20 19:19:17.000000000 +0300
@@ -28,7 +28,7 @@
 #-------------------------------------------------------------------------------
 #check if we're running on olinuxino / odroid / etc
 MESSAGE("Looking for libMali.so")
-if(EXISTS "/usr/lib/libMali.so" OR
+if(EXISTS "${CMAKE_FIND_ROOT_PATH}/usr/lib/libMali.so" OR
    EXISTS "/usr/lib/arm-linux-gnueabihf/libMali.so" OR
    EXISTS "/usr/lib/aarch64-linux-gnu/libMali.so")
     MESSAGE("libMali.so found")
