--- emulationstation-e95eb4e/es-app/src/guis/GuiMenu.cpp	2017-02-12 12:43:57.071111884 -0600
+++ emulationstation-e95eb4e/es-app/src/guis/GuiMenu.cpp	2017-02-12 14:39:30.193752024 -0600
@@ -197,16 +197,41 @@
 			);
 	});
 
-	addEntry("QUIT", 0x777777FF, true, 
+		addEntry("RETROARCH", 0x777777FF, true, 
+ 		[this] {
+						remove("/var/lock/start.games");
+						runSystemCommand("touch /var/lock/start.retro");
+						runSystemCommand("systemctl start retroarch.service");
+ 						SDL_Event ev;
+ 						ev.type = SDL_QUIT;
+ 						SDL_PushEvent(&ev);
+ 	});
+ 
+
+	addEntry("KODI", 0x777777FF, true, 
 		[this] {
+
+						remove("/var/lock/start.games");
+						runSystemCommand("touch /var/lock/start.kodi");
+						runSystemCommand("systemctl start kodi.service");
 						SDL_Event ev;
 						ev.type = SDL_QUIT;
 						SDL_PushEvent(&ev);
 	});
 
+	addEntry("REBOOT", 0x777777FF, true,
+		[this] {
+						runRestartCommand();
+	});
+
+	addEntry("SHUTDOWN", 0x777777FF, true, 
+		[this] {
+						runShutdownCommand();
+	});
+
 	mVersion.setFont(Font::get(FONT_SIZE_SMALL));
 	mVersion.setColor(0xC6C6C6FF);
-	mVersion.setText("EMULATIONSTATION V" + strToUpper(PROGRAM_VERSION_STRING));
+	mVersion.setText("EMULATIONSTATION V" + strToUpper(PROGRAM_VERSION_STRING)+" SX05RE");
 	mVersion.setAlignment(ALIGN_CENTER);
 
 	addChild(&mMenu);

--- emulationstation-e95eb4e/es-app/src/main.cpp	2017-02-01 18:20:17.000000000 -0600
+++ emulationstation-e95eb4e/es-app/src/main.cpp	2017-02-12 14:42:32.250715027 -0600
@@ -254,6 +254,9 @@
 		window.pushGui(new GuiMsgBox(&window,
 			errorMsg,
 			"QUIT", [] { 
+                                remove("/var/lock/start.games");
+				runSystemCommand("touch /var/lock/start.kodi");
+				runSystemCommand("systemctl start kodi.service");
 				SDL_Event* quit = new SDL_Event();
 				quit->type = SDL_QUIT;
 				SDL_PushEvent(quit);

--- emulationstation-e95eb4e/es-core/src/platform.cpp	2017-02-01 18:20:17.000000000 -0600
+++ emulationstation-e95eb4e/es-core/src/platform.cpp	2017-02-12 14:45:00.867871656 -0600
@@ -43,20 +43,12 @@
 
 int runShutdownCommand()
 {
-#ifdef WIN32 // windows
-	return system("shutdown -s -t 0");
-#else // osx / linux
-	return system("sudo shutdown -h now");
-#endif
+return system("systemctl poweroff");
 }
 
 int runRestartCommand()
 {
-#ifdef WIN32 // windows
-	return system("shutdown -r -t 0");
-#else // osx / linux
-	return system("sudo shutdown -r now");
-#endif
+return system("systemctl reboot");
 }
 
 int runSystemCommand(const std::string& cmd_utf8)
@@ -87,4 +79,4 @@
 	int fd = open(filename.c_str(), O_CREAT|O_WRONLY, 0644);
 	if (fd >= 0)
 		close(fd);
-}
\ No newline at end of file
+}
