--- /Source/LibPNG/pngpriv.h	2015-03-02 01:50:58.000000000 -0600
+++ /Source/LibPNG/pngpriv.h	2017-02-22 17:20:14.914196305 -0600
@@ -125,9 +125,9 @@
     */
 #  if (defined(__ARM_NEON__) || defined(__ARM_NEON)) && \
    defined(PNG_ALIGNED_MEMORY_SUPPORTED)
-#     define PNG_ARM_NEON_OPT 2
+#     define PNG_ARM_NEON_OPT 0
 #  else
-#     define PNG_ARM_NEON_OPT 0
+#     define PNG_ARM_NEON_OPT 2
 #  endif
 #endif
 
