--- a/Makefile.am	2017-09-01 14:08:53.826802649 +0000
+++ b/Makefile.am	2017-09-01 14:09:02.896683431 +0000
@@ -20,4 +20,4 @@
 #	51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 
-SUBDIRS = libexfat dump fsck fuse label mkfs
+SUBDIRS = libexfat dump fsck label mkfs
--- a/configure.ac	2017-09-01 14:08:57.016760728 +0000
+++ b/configure.ac	2017-09-01 14:09:22.226429078 +0000
@@ -31,13 +31,11 @@
 AC_PROG_RANLIB
 AM_PROG_AR
 AC_SYS_LARGEFILE
-PKG_CHECK_MODULES([FUSE], [fuse])
 AC_CONFIG_HEADERS([libexfat/config.h])
 AC_CONFIG_FILES([
 	libexfat/Makefile
 	dump/Makefile
 	fsck/Makefile
-	fuse/Makefile
 	label/Makefile
 	mkfs/Makefile
 	Makefile])
