--- a/configure	2017-05-04 04:29:56.000000000 -0500
+++ b/configure	2017-06-07 20:43:27.458122917 -0500
@@ -18286,25 +18286,7 @@
       ;;
   esac
 
-  if test x"$AR" != x; then
-    emblibav_configure_args="$emblibav_configure_args --ar=\\\"\\\$AR\\\""
-  fi
-
-  if test x"$AS" != x; then
-    emblibav_configure_args="$emblibav_configure_args --as=\\\"\\\$AS\\\""
-  fi
-
-  if test x"$CC" != x; then
-    emblibav_configure_args="$emblibav_configure_args --cc=\\\"\\\$CC\\\""
-  fi
-
-  if test x"$LD" != x; then
-    emblibav_configure_args="$emblibav_configure_args --ld=\\\"\\\$CC\\\""
-  fi
-
-  if test x"$NM" != x; then
-    emblibav_configure_args="$emblibav_configure_args --nm=\\\"\\\$NM\\\""
-  fi
+  
 
     LIBAV_OPTS=`cd $srcdir/gst-libs/ext/libav && ./configure --help`
   # Let's check if we can disable the building of the libav binary
