diff --git a/AIO/drivers/qcacld-new/CORE/HDD/src/wlan_hdd_main.c b/AIO/drivers/qcacld-new/CORE/HDD/src/wlan_hdd_main.c
index 8579ab8..fb58f83 100644
--- a/AIO/drivers/qcacld-new/CORE/HDD/src/wlan_hdd_main.c
+++ b/AIO/drivers/qcacld-new/CORE/HDD/src/wlan_hdd_main.c
@@ -253,7 +253,7 @@ static VOS_STATUS hdd_parse_ese_beacon_req(tANI_U8 *pValue,
  #define A_MEMCMP(addr1, addr2, len)     memcmp((addr1), (addr2), (len))
  #define A_ROUND_UP(x, y)  ((((x) + ((y) - 1)) / (y)) * (y))
  
- char qcafwpath[256] = "/system/etc/wifi/firmware";
+ char qcafwpath[256] = "/lib/firmware/qca9377";
  
  int android_readwrite_file(const A_CHAR *filename, A_CHAR *rbuf, const A_CHAR *wbuf, size_t length)
  {
